core/src/main/scala/scala/pickling/spi/PicklerRegistry.scala

```diff
@@ -56,23 +56,22 @@ trait PicklerRegistry {
     *             In those situations, the pickler should be able to handle arbitrary (existential) type parameters.
     * @param p  The pickler to register.
     */
-  def registerPickler[T](key: String, p: Pickler[T]): Unit
-  def registerPickler[T](p: Pickler[T]): Unit = registerPickler(p.tag.key, p)
+  def registerPickler(key: String, p: Pickler[_]): Unit
+  def registerPickler(p: Pickler[_]): Unit = registerPickler(p.tag.key, p)
```

Comment:
The problem here is that we do not use those type parameters for nothing. Internally, the maps are storing picklers ignoring the type (using existential qualifiers) and therefore to have them there is a problem for `dumpStateTo` when you want to register again a pickler/unpickler. Of course one could use `Any` as the type, but it doesn't feel right.